src/net/udp.rs

```diff
@@ -80,22 +80,35 @@ impl UdpSocket {
         if let Some(link_conditioner) = &self.link_conditioner {
             if link_conditioner.should_send() {
                 for payload in packet_data.parts() {
-                    bytes_sent += self.socket.send_to(&payload, packet.addr()).map_err(|io| {
-                        NetworkError::from(NetworkErrorKind::IOError { inner: io })
-                    })?;
+                    bytes_sent += self.send_packet(&packet.addr(), payload)?;
                 }
             }
         } else {
+            if let Some(dropped) = lock.gather_dropped_packets() {
+                for payload in dropped {
```

Comment:
as I said about you just need to create an `iter` here for dropped packets. if the iter is empty it doesn't run the code within the loop.

Reply:
Done